source/common/network/dns_impl.h

```diff
@@ -1,47 +1,68 @@
 #pragma once
 
+#include "envoy/event/file_event.h"
 #include "envoy/network/dns.h"
 
 #include "common/common/linked_object.h"
 #include "common/event/dispatcher_impl.h"
 
+#include "ares.h"
 #include "event2/event_struct.h"
 
 namespace Network {
 
+class DnsResolverImplPeer;
```

Comment:
At Google, we tend to use this FooPeer pattern for reaching into the implementation internals. I'll look at refactoring, but it seems a useful tool to have available when writing tests in general, avoiding introducing test specific behavior into the implementation.